### PR TITLE
feat: tslint rules enable

### PR DESCRIPTION
**Pull request by aashishmalik**
## Description of the change

> tslint rules enable in tranformer

> Change type 'Object' to 'object` . 
Ref.  `error`  Don't use `Object` as a type.
- If you want a type meaning "any object", you probably want `object` instead.
- If you want a type meaning "any value", you probably want `unknown` instead.

> Create a type for `any` as wasn't sure on the structure, to remove lint error on further code added to repository . 

## Type of change
- [ ] Bug fix (non-breaking change that fixes an issue)
- [ ] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Related issues

> Fix [#1]() 

### Development

- [ ] Lint rules pass locally
- [ ] The code changed/added as part of this pull request has been covered with tests
- [ ] All tests related to the changed code pass in development

### Code review 

- [ ]  This pull request has a descriptive title and information useful to a reviewer. There may be a screenshot or screencast attached
- [ ] "Ready for review" label attached to the PR and reviewers mentioned in a comment
- [ ] Changes have been reviewed by at least one other engineer
- [ ] Issue from task tracker has a link to this pull request

### CODE REVIEW

**Review by koladilip — DISMISSED** [review]

File: src/controllers/bulkUpload.ts [hunk 1]
```diff
@@ -42,16 +56,16 @@ export const fileUpload = async (ctx) => {
   let response;
   try {
     response = await destFileUploadHandler.processFileData(ctx.request.body);
-  } catch (error: any) {
+  } catch (error: CatchErr) {
     response = {
-      statusCode: error?.response?.status || error?.status || 400,
-      error: error.message || 'Error occurred while processing payload.',
+      statusCode: error.response ? error.response.status : 400,
```
Comment (sanpj2292) [inline src/controllers/bulkUpload.ts:61]: `error.status` is not required at all ? Did we confirm about this ?

File: src/controllers/profile.ts [hunk 2]
```diff
@@ -30,7 +29,6 @@ export default class ProfileController {
       }'
   */
   public static async profile(ctx: Context) {
-    const requestMetadata = MiscService.getRequestMetadata(ctx);
```
Comment (sanpj2292) [inline src/controllers/profile.ts:33]: Did we remove because it was un-used ?

File: src/middlewares/routeActivation.ts [hunk 3]
```diff
@@ -78,6 +78,7 @@ export default class RouteActivationMiddleware {
     );
   }
 
+  // eslint-disable-next-line sonarjs/no-identical-functions
```
Comment (sanpj2292) [inline src/middlewares/routeActivation.ts:81]: Why are we adding this ?

File: src/services/source/nativeIntegration.ts [hunk 4]
```diff
@@ -9,6 +9,8 @@ import PostTransformationServiceSource from './postTransformation';
 import FetchHandler from '../../helpers/fetchHandlers';
 import tags from '../../v0/util/tags';
 import stats from '../../util/stats';
+// eslint-disable-next-line @typescript-eslint/no-unused-vars
```
Comment (sanpj2292) [inline src/services/source/nativeIntegration.ts:12]: Why is this line added ? I'm seeing we are using CatchErr and FixMe in this file

File: src/testRouter.js [hunk 5]
```diff
@@ -28,8 +28,8 @@ const isSupportedContentType = (contentType) => {
 
 const testRouter = new Router({ prefix: '/test-router' });
 
-const getDestHandler = (version, destination) =>
-  require(`./${version}/destinations/${destination}/transform`);
+const getDestHandler = (versionNo, destination) =>
```
Comment (sanpj2292) [inline src/testRouter.js:31]: Currently version is not a number but rather a string as `v0`. So I don't think the variable `versionNo` makes sense. Let me know if you think the same.

File: src/v0/destinations/active_campaign/transform.js [hunk 6]
```diff
@@ -337,7 +337,8 @@ const customListProcessor = async (message, category, destination, contactId) =>
   // status information
   // Ref: https://developers.activecampaign.com/reference/update-list-status-for-contact/
   const promises = [];
-  listArr.map(async (li) => {
+  // eslint-disable-next-line no-restricted-syntax
+  for (const li of listArr) {
```
Comment (sanpj2292) [inline src/v0/destinations/active_campaign/transform.js:341]: Instead of `for...of` loop, we can use map function to put things into promise
```javascript
const promises = listArr.map(async (li) => {
  return someVar;
})
.filter(v => undefined) // We would need to test this in order to understand how we remove undefined promises
```
This should be preferred approach

File: src/controllers/bulkUpload.ts (hunk 1 above)
Comment (aashishmalik) [reply src/controllers/bulkUpload.ts:61]: reverted

File: src/testRouter.js (hunk 5 above)
Comment (aashishmalik) [reply src/testRouter.js:31]: changed to ver, version is giving lint issue as version is declared in outer scope

File: src/middlewares/routeActivation.ts (hunk 3 above)
Comment (aashishmalik) [reply src/middlewares/routeActivation.ts:81]: there were two different functions which were identical but with different fn names in two different flows, so going with current implementation of maintaining two different fn names with same body to accommodate changes in 1 flow if comes later

File: src/v0/destinations/active_campaign/transform.js (hunk 6 above)
Comment (aashishmalik) [reply src/v0/destinations/active_campaign/transform.js:341]: I will take `no-restricted-syntax` in another PR, as changes introduced with new develop merge results these in different files. For now going with disabling the rule

**Review by sanpj2292 — APPROVED** [review]

File: src/v0/destinations/mp/util.js [hunk 7]
```diff
@@ -143,6 +143,7 @@ const isImportAuthCredentialsAvailable = (destination) =>
 const findExistingBatch = (batch, metadataMap) => {
   let existingBatch = null;
 
+  // eslint-disable-next-line no-restricted-syntax
```
Comment (Gauravudia) [inline src/v0/destinations/mp/util.js:146]: can you resolve this? We need to disable the eslint rule on next line?

**Review by koladilip — APPROVED** [review]

File: src/v0/destinations/mp/util.js (hunk 7 above)
Comment (aashishmalik) [reply src/v0/destinations/mp/util.js:146]: the existing code written with `for of` loop was shoeing this error, needed to update that code . Planning to take these in different PRas develop merge results in this errors for differnt files as well
